deepchem/models/tests/test_layers.py

```diff
@@ -943,3 +944,25 @@ def test_local_global_discriminator():
     # Check if the output has the correct shape and dtype
     assert similarity_scores.shape == (batch_size,)
     assert similarity_scores.dtype == torch.float32
+
+
+@pytest.mark.torch
+def test_set_gather():
+    """Test invoking the Torch Equivalent of SetGather."""
+    # total_n_atoms = 4
+    # n_atom_feat = 4
+    # atom_feat = np.random.rand(total_n_atoms, n_atom_feat)
+    atom_feat = np.load(
+        os.path.join(os.path.dirname(__file__), "assets",
+                     "atom_feat_SetGather.npy"))
+    atom_split = np.array([0, 0, 1, 1], dtype=np.int32)
+    torch_layer = torch_layers.SetGather(2, 2, 4)
+    weights = np.load(
+        os.path.join(os.path.dirname(__file__), "assets",
+                     "weights_SetGather_tf.npy"))
+    torch_layer.U = torch.nn.Parameter(torch.from_numpy(weights))
```

Comment:
is the .U attribute meaningful in pytorch? Not sure if this is necessary. Check if it's needed, if not, remove it. Otherwise, explain why it's needed.

Reply:
Its required so not removing.